### PR TITLE
feat(universes): Add place-level user restriction support

### PR DESCRIPTION
**Pull request by commonly-ts**
# Description

Extends the Universes class to allow updating user restrictions at the place level by accepting an optional placeId parameter. Adjusts the resource path logic to handle both universe-level and place-level user restrictions.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [x] ✅ Test update

## Changes Made

- Adds `placeId` parameter to `universes.updateUserRestriction`
- Updates `universes.updateUserRestriction` to use an options object for `universeId`, `placeId` and `body`.
- Adds placeId check to the updateUserRestriction test.

## Testing

- [x] All existing tests pass (`pnpm test`)
- [ ] Added new tests for new functionality
- [ ] Tested manually (describe below)

### Manual Testing

Simple string concatenation so no manual testing needed

## Code Quality

- [x] My code follows the project's coding style
- [x] I have run `pnpm lint` and fixed any issues
- [x] I have run `pnpm format` to format my code
- [x] I have run `pnpm typecheck` and there are no type errors
- [x] I have reviewed my own code

## Documentation

- [ ] I have updated the relevant documentation (if applicable)
- [x] I have added/updated JSDoc comments for new/modified code

## Screenshots/Recordings

image

## Checklist

- [x] I have read the [CONTRIBUTING.md](../CONTRIBUTING.md) guide
- [x] My PR title follows the [Conventional Commits](https://www.conventionalcommits.org/) format
- [x] I have rebased my branch on the latest `main` branch
- [ ] I have tested my changes against the Roblox Open Cloud API (if applicable)
- [x] This PR is ready for review

---

### CODE REVIEW

**Comment by commonly-ts** [comment]
I would agree if the method's parameters are a bit icky now, I'm fine with changing it into options typed with an interface if you'd like

**Review by marinofranz — CHANGES_REQUESTED** [review]
I agree, let's make this into an options object so we can re-use
There are a couple of other methods that have this feature as well, you can integrate the options object into that as well if you're feeling up for it 🙂 

Make sure to add this to a test or create a new one so coverage can pass

File: src/resources/universes.ts [hunk 1]
```diff
@@ -156,33 +155,47 @@ export class Universes {
...
+   *    excludeAltAccounts: true
+   *  }
    * });
    * ```
    *
    * @see https://create.roblox.com/docs/cloud/reference/UserRestriction#Cloud_UpdateUserRestriction__Using_Universes_Places
    */
   async updateUserRestriction(
-    universeId: string,
     userRestrictionId: string,
```
Comment (marinofranz) [inline src/resources/universes.ts]: Swap `userRestrictionId` with `universeId` and update the payload so it contains the `useRestrictionId` instead
This way, we stick to the code style of always specifying the `universeId` in the first parameter.

File: src/resources/universes.ts [hunk 2]
```diff
@@ -156,33 +155,47 @@ export class Universes {
...
-    const searchParams = new URLSearchParams();
+    const { body, universeId, placeId } = options;
+    const payload = { ...body };
 
+    // API returns 400 if duration is set as an empty string
+    if (payload.duration == "") {
+      console.log(payload);
+      payload.duration = undefined;
+      console.log(payload);
+    }
```
Comment (marinofranz) [inline src/resources/universes.ts]: Refactor this to make it cleaner (remove the entire if statement)
```typescript
payload.duration = payload.duration === "" ? undefined : payload.duration;

```

File: src/resources/universes.ts [hunk 3]
```diff
@@ -156,33 +155,47 @@ export class Universes {
...
    */
   async updateUserRestriction(
-    universeId: string,
     userRestrictionId: string,
-    body: GameJoinRestriction,
+    options: UpdateUserRestrictionOptions,
   ): Promise<UserRestriction> {
-    const searchParams = new URLSearchParams();
+    const { body, universeId, placeId } = options;
+    const payload = { ...body };
```
Comment (marinofranz) [inline src/resources/universes.ts]: Is the shallow copy of `body` required here? I'm okay with this as long as it's got a reasonable explanation

File: src/resources/universes.ts [hunk 4]
```diff
@@ -191,14 +204,20 @@ export class Universes {
...
-      },
-    );
+    let resourcePath = `/cloud/v2/universes/${universeId}`;
+
+    // This resource has two different paths, one for universe-level restrictions, and another for place-level restrictions
+    if (placeId) {
+      resourcePath += `/places/${placeId}`;
+    }
+
+    resourcePath += `/user-restrictions/${userRestrictionId}`;
```
Comment (marinofranz) [inline src/resources/universes.ts:203]: We can replace this with an array then removing any items that are falsey
```typescript
const resourcePath = [
  `/cloud/v2/universes/${universeId}`,
  placeId ? `/places/${placeId}` : null,
  `/user-restrictions/${userRestrictionId}`
].filter(Boolean).join("");
```

File: src/resources/universes.ts [hunk 5]
```diff
@@ -191,14 +204,20 @@ export class Universes {
...
+
+    return this.http.request<UserRestriction>(resourcePath.toString(), {
```
Comment (marinofranz) [inline src/resources/universes.ts]: `.toString()` not required here since `resourcePath` is already a string

**Review by marinofranz — CHANGES_REQUESTED** [review]
There are a few changes that I want to address regarding the code style, the logic is perfectly fine 🙂

File: src/resources/universes.ts (hunk 3 above)
Comment (commonly-ts) [reply src/resources/universes.ts]: oh I forgot to remove the stuff I was debugging the test with 💔

File: src/resources/universes.ts (hunk 4 above)
Comment (commonly-ts) [reply src/resources/universes.ts:203]: I do feel the if statement for the resourcePath is much more readable

**Comment by commonly-ts** [comment]
also wondering about readability for `duration` as its not particularly clear that it needs to be in seconds with "s" at the end, unless you go and read the API docs.

could rename it to durationSeconds and have it as a number, and then parse it inside the function or is this SDK just a base wrapper without that sort of thing?

**Comment by marinofranz** [comment]
> also wondering about readability for `duration` as its not particularly clear that it needs to be in seconds with "s" at the end, unless you go and read the API docs.
> 
> could rename it to durationSeconds and have it as a number, and then parse it inside the function or is this SDK just a base wrapper without that sort of thing?

We try to stick to just the API reference but we can eventually figure out a helper function for that, this will do for now 🙂 
There's just one missing test that should test if `duration` is an empty string to pass coverage

File: src/resources/universes.ts (hunk 4 above)
Comment (marinofranz) [reply src/resources/universes.ts:203]: Fine with me, it might be changed in the future, it's undecided for now
